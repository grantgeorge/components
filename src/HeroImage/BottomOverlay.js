// @flow

import styled from 'styled-components'

const BottomOverlay = styled.div`
  position: absolute;
  height: 100%;
  width: 100%;
  bottom: 0;
  background-image: linear-gradient(
    to bottom,
    rgba(0, 0, 0, 0) 70%,
    rgba(0, 0, 0, 0.6)
  );
`

export default BottomOverlay
