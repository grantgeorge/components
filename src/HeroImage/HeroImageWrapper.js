// @flow

import styled from 'styled-components'

const HeroImageWrapper = styled.div`
  margin: 0;
  box-shadow: 0 3px 6px 1px rgba(0, 0, 0, 0.16),
    0 3px 6px 1px rgba(0, 0, 0, 0.23);
`

export default HeroImageWrapper
