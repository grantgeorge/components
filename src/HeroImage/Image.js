// @flow

import styled from 'styled-components'

const HeroImage = styled.div`
  background-repeat: no-repeat;
  background-position: center center;
  background-size: cover;
  position: relative;
  width: 100%;
  height: 170px;
  border: none;
  background-color: transparent;
  background-image: url(${(props) => props.imageUrl});

  @media (min-width: 568px) {
    height: 400px;
  }
`

export default HeroImage
