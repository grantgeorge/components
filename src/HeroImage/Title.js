// @flow

import styled from 'styled-components'

const Title = styled.div`
  position: absolute;
  color: #fff;
  font-weight: 800;
  line-height: inherit;
  left: 0;
  bottom: 0;
  width: 100%;
  text-align: center;
  font-size: 2.2rem;

  @media (min-width: 768px) {
    line-height: 1;
    font-size: 4em;
    bottom: 15px;
    left: 25px;
    width: inherit;
    text-align: inherit;
  }
`

export default Title
