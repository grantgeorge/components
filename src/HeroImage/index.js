// @flow

/**
 *
 * HeroImage
 *
 */

import React from 'react'

import BottomOverlay from './BottomOverlay'
import Image from './Image'
import HeroImageWrapper from './HeroImageWrapper'
import Title from './Title'

type Props = {
  title: string,
  imageUrl: string,
}

const HeroImage = ({ title, imageUrl }: Props) => {
  return (
    <HeroImageWrapper>
      <Image imageUrl={imageUrl}>
        <BottomOverlay />
        <Title>{title}</Title>
      </Image>
    </HeroImageWrapper>
  )
}

export default HeroImage
