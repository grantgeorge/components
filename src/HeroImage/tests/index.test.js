import React from 'react'
import { shallow } from 'enzyme'

import HeroImage from '../index'

describe('<HeroImage />', () => {
  let renderedComponent

  beforeEach(() => {
    renderedComponent = shallow(<HeroImage />)
  })

  it('should render properly', () => {
    expect(renderedComponent).toMatchSnapshot()
  })
})
