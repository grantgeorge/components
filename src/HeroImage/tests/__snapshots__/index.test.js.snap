// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroImage /> should render properly 1`] = `
<styled.div
  style={Object {}}
>
  <styled.div>
    <styled.div />
  </styled.div>
</styled.div>
`;
